src/main/java/me/karakelley/http/Request.java

```diff
@@ -0,0 +1,62 @@
+package me.karakelley.http;
+
+import java.io.BufferedReader;
+import java.util.HashMap;
+
+public class Request {
+  private final RequestLine requestLine;
+  private final RequestHeaders requestHeaders;
+  private final BufferedReader reader;
+
+  public Request(BufferedReader reader) {
```

Comment:
Why depend on such a concrete class? Would `InputStream` suffice?

Reply:
I was depending on `BufferedReader` in order to use `readLine()`